elegant/segment_images.py

```diff
@@ -192,7 +192,7 @@ def find_lawn_in_image(image, optocoupler, return_model=False):
 
 def _get_pose(mask, timepoint_annotations, timepoint_name, width_estimator):
     center_tck, width_tck = worm_spline.pose_from_mask(mask)
-    if width_estimator is not None:
+    if width_estimator is not None and center_tck is not None:
```

Comment:
since `width_tck` is what's mutated, you should test it for being none, not `center_tck`. There currently aren't circumstances where one is `None` but the other isn't, but that doesn't mean this won't change in the future. General rule of thumb: tests like this should be directly for the thing you want, not for something else that is correlated with that thing.

Reply:
resolved